boards: Introduce AI-WB2-12F

VynDragon opened the pull request:
Introduces the Ai-Thinker WB2 development board.

depends on #89021

VynDragon committed:
boards: bflb: Introduce AI-WB2-12F

This introduces the Ai-Thinker BL602 WB2 development board

Signed-off-by: Camille BAUD <mail@massdriver.space>

VynDragon committed:
dts: bindings: Add Zbit Semiconductor

Add zb prefix

Signed-off-by: Camille BAUD <mail@massdriver.space>

nordicjm reviewed (dismissed):
binding update should be before dts that uses it but likely not an issue with CI

rruuaanng commented on boards/aithinker/ai_wb2_12f/ai_wb2_12f.dts:
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2024-2025 MASSDRIVER EI (massdriver.space
```
```suggestion
 * Copyright (c) 2024-2025 MASSDRIVER EI (massdriver.space)
```

VynDragon replied:
Done

VynDragon commented:
Marking DNM until https://github.com/zephyrproject-rtos/zephyr/pull/89021 is solved due to likely requiring some path changes.

VynDragon commented:
Sorry for rebase, I only changed `#include <bouffalolab/bl60x.dtsi>` to `#include <bflb/bl60x.dtsi>`

VynDragon commented:
Ci fail not related to pr.

kartben commented on boards/aithinker/ai_wb2_12f/doc/index.rst:
Please check the contents of this file ;)

VynDragon replied:
I dont see anything wrong with it?

VynDragon replied:
You want removal of `The default configuration can be found in the Kconfig
:zephyr_file:`boards/aithinker/ai_wb2_12f/ai_wb2_12f_defconfig`. ?` But i looked at th doc build and really dont see what's wrong.

kartben replied:
Yeah that would be great to drop for sure.
And sorry for not having been more clear -- some copy paste leftovers from bl604 dvk "jumped" at me hence why I commented with a brief "please double check this file". Sorry for the confusion .

VynDragon replied:
Well yea, it's basically the same thing with less pins, not much else to say, maybe some talk about the 12F format?

kartben commented on boards/aithinker/ai_wb2_12f/support/openocd.cfg:
```diff
@@ -0,0 +1,5 @@
+# For WCH linkE in DAP mode
```
WCH?

VynDragon replied:
Yes, WCH, more convenient than the RVDebugger (bl702) or ftdi 232hq.

kartben commented on boards/aithinker/ai_wb2_12f/doc/index.rst:
```diff
@@ -0,0 +1,108 @@
+various security features.  The external interfaces include SDIO, SPI, UART,
+I2C, IR remote, PWM, ADC, DAC, PIR and GPIO.
+
+The BL602 Development Board features a SiFive E24 32 bit RISC-V CPU with FPU,
```
E.g this

kartben replied:
I don't mind the copy-paste of the blurb but just fix the name of the board :)

VynDragon replied:
do you mean 'The BL602 Development Board' ?
This BL602 Development Board? The BL602 WB2 12F Module Development Board?

kartben commented on boards/aithinker/ai_wb2_12f/ai_wb2_12f_defconfig:
```diff
@@ -0,0 +1,9 @@
+CONFIG_SERIAL=y
+
+CONFIG_UART_CONSOLE=y
+CONFIG_UART_BFLB=y
```
```suggestion
```

VynDragon replied:
Cant, 'default y' is missing from the KConfig.

VynDragon replied:
I added a commit to fix it actually.

kartben replied:
Cool, thanks! You should also drop it from bl604dvk defconfig while you're here.

VynDragon replied:
Done

VynDragon committed:
drivers: serial: add missing default y

Adds missing default y to bflb Kconfig

Signed-off-by: Camille BAUD <mail@massdriver.space>

VynDragon committed:
boards: bl604e_iot_dvk: remove CONFIG_UART_BFLB

remove CONFIG_UART_BFLB because default y was added to
Kconfig so it enables automatically

Signed-off-by: Camille BAUD <mail@massdriver.space>

VynDragon commented:
Rebased on main. Also CI keeps getting stuck, is devops investigating this?

kartben commented:
-

VynDragon commented:
Iame not able to

nandojve commented:
Maybe a force push could re-open.

nandojve commented:
re-open.

nandojve commented:
> re-open.

fumy, it showed to me to re-open with a message but did not performed : (

VynDragon commented:
Tried pushing a rebase, and modifying commit message...

VynDragon commented:
ungreyed Reopen button test 4

VynDragon commented:
test 5

VynDragon commented on boards/aithinker/ai_wb2_12f/board.yml:
```diff
@@ -0,0 +1,6 @@
+board:
+  name: ai_wb2_12f
+  full_name: Ai-Thinker WB2-12F development board
+  vendor: bflb
```
it should probably also added in the vendor-prefixes list, is that correct?

kartben replied:
Which you've already done in this PR :)

kartben replied:
Or so I thought, sorry
